### ISSUE TITLE
Newest version(origin/master) memory issue on GPU

### ISSUE THREAD

**Issue by shwoo93** [pytorch/pytorch#2187]
I have worked for pytorch with official released version but there were some limits related to GPU. For example, executing torch.range() on GPU gave me the below error, 

**torch.cuda.FloatTensor doesn't implement stateless method arange**

Therefore I just cloned pytorch git and build the docker image from the provided dockerfile.
Finally, the code execution of torch.range() on GPU was not the problem but I got an another issue.

**RuntimeError: cuda runtime error (2) : out of memory at /tmp/pip-ikrnhryb-build/torch/lib/THC/generic/THCStorage.cu:66**

Firstly, I tried to change batch_size from 16 to 8 but gave me the same error. 
I tried batch_size 2 and 1 as well  but result in with the below errors respectively.

**invalid argument 2: sizes do not match at /tmp/pip-ikrnhryb-build/torch/lib/THC/generated/../generic/TH  --> batch_size = 2**

**assert len(modules) == len(devices) --> batch_size = 1**

Since now there is a memory issue with the latest version, I am wondering if it will take some times to be completely stable. Or is there any way to resolve this situation?

Need your help!

**Comment by soumith** [comment]
can you give me a code snippet to reproduce this out of memory (that didnt run out of memory in 0.1.12)

**Comment by fmassa** [comment]
Did you make sure you compiled with CUDNN support? If not, that could explain the differences in memory utilisation.

**Comment by ngimel** [comment]
Both docker image and binary packages are with cudnn support, so that should not be the issue.

**Comment by shwoo93** [comment]
![image](https://user-images.githubusercontent.com/13035722/29012339-d5b05d74-7b75-11e7-84a2-d7602bb49d7d.png)

**Comment by shwoo93** [comment]
This is the error what I receive continuously from the code when I use the newest version of pytorch

**Comment by alykhantejani** [comment]
@shwoo93 thanks. Is it possible to give us a snippet of code that will produce this issue in master and not in 0.1.12?

**Comment by soumith** [comment]
@shwoo93 still waiting on this. Please open the issue again with code once you provide it. Thank you.
